pkg/tagger/types/types.go

```diff
@@ -7,21 +7,22 @@
 package types
 
 // ProductOrigin is the origin of the product that sent the entity.
-type ProductOrigin string
+type ProductOrigin int
 
 const (
 	// ProductOriginDogStatsD is the ProductOrigin for DogStatsD.
-	ProductOriginDogStatsD ProductOrigin = "dogstatsd"
+	ProductOriginDogStatsD ProductOrigin = iota
 	// ProductOriginAPM is the ProductOrigin for APM.
-	ProductOriginAPM ProductOrigin = "apm"
+	ProductOriginAPM ProductOrigin = iota
 )
 
 // OriginInfo contains the Origin Detection information.
 type OriginInfo struct {
-	FromUDS       string
-	FromTag       string
-	FromMsg       string
-	Cardinality   string
-	ProductOrigin ProductOrigin
-	OptOutEnabled *bool
+	FromUDS                   string        // FromUDS is the origin resolved using Unix Domain Socket.
+	FromTag                   string        // FromTag is the origin resolved from tags.
+	FromMsg                   string        // FromMsg is the origin resolved from the message.
+	Cardinality               string        // Cardinality is the cardinality of the resolved origin.
+	ProductOrigin             ProductOrigin // ProductOrigin is the product that sent the origin information.
+	OptOutEnabled             *bool         // OptOutEnabled is a pointer to a boolean indicating if the origin opt-out is enabled.
+	EntityIDPrecedenceEnabled *bool         // EntityIDPrecedenceEnabled is a pointer to a boolean indicating if EntityIDPrecedence is enabled.
```

Comment:
These two are static configuration options, I don't think we need to copy them to every metric sample we handle, which costs us extra memory usage. Would it be possible to read the config settings in the tagger, or pass the values as parameters?

Reply:
I changed the code to read the setting directly from the Tagger 👍

Reply:
I changed the code to read the setting directly from the Tagger 👍